tests/snuba/search/test_backend.py

```diff
@@ -1217,6 +1229,245 @@ def test_first_release_environments(self):
         )
         assert set(results) == set([self.group1])
 
+    def test_first_release_any_or_no_environments(self):
```

Comment:
I like that this test is pretty thorough, although it is very verbose. Is there anything you can do to get the size down?

Just as a reference, this entire file is currently 1400 lines, and this test is about ~250 lines just by itself. I'm concerned that it's going to be hard to maintain in the future because it's just a lot to read.

Reply:
@wedamija I can definitely understand the concern that the test is quite long. I tried to write out comments to segment sections of the tests. 

Though, I'm curious, is the entire test function not readable despite having been documented?  🤔 

About ~50 loc are comments at the beginning of the function.

------

I'm open to more suggestions on what can be bikeshedded.